.github/workflows/upload-and-deploy.yaml

```diff
@@ -48,21 +48,60 @@ jobs:
         uses: aws-actions/configure-aws-credentials@v1
         with:
           aws-region: us-east-1
-          role-to-assume: arn:aws:iam::914833433684:role/sagebase-github-oidc-sage-bion-ProviderRolerecover-VTOKECAKJUBV
+          role-to-assume: ${{ vars.AWS_CREDENTIALS_IAM_ROLE }}
           role-session-name: GitHubActions-${{ github.repository_owner }}-${{ github.event.repository.name }}-${{ github.run_id }}
           role-duration-seconds: 1200
 
       - name: Setup sam
         uses: aws-actions/setup-sam@v2
 
       - name: Set namespace for non-default branch or for tag
-        if: github.ref_name != 'main'
         run: echo "NAMESPACE=$GITHUB_REF_NAME" >> $GITHUB_ENV
 
-      - name: Copy files to templates bucket, use dev cloudformation bucket if not main branch
-        if: github.ref_name != 'main'
-        run: python src/scripts/manage_artifacts/artifacts.py --upload --namespace $NAMESPACE --cfn_bucket recover-dev-cloudformation
+      - name: Copy files to templates bucket, use dev cloudformation bucket
+        run: python src/scripts/manage_artifacts/artifacts.py --upload --namespace $NAMESPACE --cfn_bucket ${{ vars.CFN_BUCKET }}
+
+
+  upload-files-main:
+    name: Upload files to S3 bucket in production
+    runs-on: ubuntu-latest
+    needs: pre-commit
+    if: github.ref_name == 'main'
+    # These permissions are needed to interact with GitHub's OIDC Token endpoint.
+    permissions:
+      id-token: write
+      contents: read
+    environment: prod
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Set up python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
+
+      - name: Install pipenv
+        shell: bash
+        run: python -m pip install pipenv
+
+      - name: Install dependencies
+        env:
+          PIPENV_NOSPIN: 'true'
+        shell: bash
+        run: |
+          pipenv install --dev
+
+      - name: configure aws credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-region: us-east-1
+          role-to-assume: ${{ vars.AWS_CREDENTIALS_IAM_ROLE }}
+          role-session-name: GitHubActions-${{ github.repository_owner }}-${{ github.event.repository.name }}-${{ github.run_id }}
```

Comment:
Nit: I think eventually - we actually want to update this: https://github.com/Sage-Bionetworks/action-pipenv-aws-setup to use OIDC integration.  That may be cleaner than all the duplicated code we have.

Lets create a ticket for this?

Reply:
Done: [IBCDPE-516](https://sagebionetworks.jira.com/browse/IBCDPE-516)